### PR TITLE
[rust] Harden processors logging

### PR DESCRIPTION
**Pull request by rtso**
* Add more logging to processors code that include tps, kbps, duration, etc. [Logging spec](https://www.notion.so/aptoslabs/Indexer-Metrics-Logging-Spec-d60a5ed2c22e4d2a9910045e3e7283b7)
* Add a timeout to the consumer thread to panic if it doesn't receive any transactions in 15s. 

Open to suggestions to make the logging cleaner. 

## Testing
Consumer thread panicked 
Screenshot 2023-11-06 at 12 22 35 PM

`cargo run --release -- -c config.yaml --verbose true`
Screenshot 2023-11-06 at 12 24 39 PM

connection id string is logged
Screenshot 2023-11-06 at 12 22 47 PM

Prometheus metrics
Screenshot 2023-11-09 at 12 45 28 PM

### CODE REVIEW

File: rust/processor/src/utils/util.rs [hunk 1]
```diff
@@ -413,6 +414,41 @@ pub fn get_name_from_unnested_move_type(move_type: &str) -> &str {
...
+// }
+
+// pub fn calculate_size_in_kbs(vectors: Vec<Box<dyn std::fmt::Debug>>) {
+//     vectors
+//         .iter()
+//         .fold(0.0, |acc, v| acc + mem::size_of_val(&v[0]) * v.len())
+// }
+
+#[macro_export]
+macro_rules! calculate_size_in_kbs {
```
Comment (bowenyang007) [inline rust/processor/src/utils/util.rs:436]: provide some reference on where this comes from? Is this chatgpt or stack overflow?

File: rust/processor/src/utils/util.rs [hunk 2]
```diff
@@ -413,6 +414,41 @@ pub fn get_name_from_unnested_move_type(move_type: &str) -> &str {
...
+//     let total_size_in_bytes = mem::size_of_val(&vector[0]) * vector.len();
+//     let total_size_in_kb = total_size_in_bytes as f64 / 1024.0;
+//     total_size_in_kb
+// }
+
+// pub fn calculate_size_in_kbs(vectors: Vec<Box<dyn std::fmt::Debug>>) {
+//     vectors
+//         .iter()
+//         .fold(0.0, |acc, v| acc + mem::size_of_val(&v[0]) * v.len())
+// }
```
Comment (bowenyang007) [inline rust/processor/src/utils/util.rs:433]: remove

File: rust/server-framework/src/lib.rs [hunk 3]
```diff
@@ -17,6 +17,8 @@ use warp::{http::Response, Filter};
 pub struct ServerArgs {
     #[clap(short, long, value_parser)]
     pub config_path: PathBuf,
+    #[clap(short, long)]
+    pub verbose: bool,
```
Comment (rtso) [inline rust/server-framework/src/lib.rs:21]: How can I update server-framework to work with the flag?

File: rust/processor/src/config.rs [hunk 4]
```diff
@@ -21,11 +21,12 @@ pub struct IndexerGrpcProcessorConfig {
     pub starting_version: Option<u64>,
     pub ending_version: Option<u64>,
     pub number_concurrent_processing_tasks: Option<usize>,
+    // pub enable_verbose_logging: Option<bool>,
```
Comment (bowenyang007) [inline rust/processor/src/config.rs:24]: add verbose here lol

Comment (rtso) [reply rust/processor/src/config.rs:24]: oh i guess we don't need to because it's passed as a param in the `run` function

Comment (rtso) [reply rust/processor/src/config.rs:24]: is that the right way to do it @larry-aptos ? i'm not completely sure how server-framework works

Comment (bowenyang007) [reply rust/processor/src/config.rs:24]: Don't pass it in the run function lmao. This config struct exists so that run doesn't have to change.

**Comment by bowenyang007** [comment]
In the summary can you also add all the field names you're using? So that we can have the same exact field names upstream.

File: rust/processor/src/config.rs (hunk 4 above)
Comment (rtso) [reply rust/processor/src/config.rs]: hmm this is gonna be hard bc it's a command line argument rather than a config.yaml field.

**Comment by larry-aptos** [comment]
1. you can do a follow-up to parse the `x-aptos-conneciton-id` in the response metadata and log it.
2. there are some fields in your screenshot but i didn't find them in the PR; anything i missed.

Overall looks good!

**Review by larry-aptos — APPROVED** [review]

File: rust/processor/src/utils/counters.rs [hunk 5]
```diff
@@ -100,16 +100,76 @@ pub static LATEST_PROCESSED_VERSION: Lazy<IntGaugeVec> = Lazy::new(|| {
...
+
+/// Overall processing time for multiple (n = number_concurrent_processing_tasks) batch of transactions
+pub static MULTI_BATCH_PROCESSING_TIME_IN_SECS: Lazy<GaugeVec> = Lazy::new(|| {
+    register_gauge_vec!(
+        "indexer_processor_multi_batch_processing_time_in_secs",
+        "Time taken to process multiple batches of transactions",
+        &["processor_name"]
+    )
+    .unwrap()
+});
```
Comment (bowenyang007) [inline rust/processor/src/utils/counters.rs:146]: can we follow the same format here? i.e. duration with processor, step, and message as keys?

Comment (bowenyang007) [reply rust/processor/src/utils/counters.rs:146]: nvm this is fine.

**Pull request merged**